src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/IncrementalBuildTest.cs

```diff
@@ -717,5 +714,26 @@ public void InvalidAndroidResource ([Values (true, false)] bool useAapt2)
 				Assert.IsFalse (b.Build (proj), "Build should *not* have succeeded.");
 			}
 		}
+
+		[Test]
+		public void CasingOnJavaLangObject ()
+		{
+			var className = "Foo";
+			var proj = new XamarinAndroidApplicationProject {
+				Sources = {
+					new BuildItem ("Compile", "Foo.cs") {
+						TextContent = () => {
+							return $"public class {className} : Java.Lang.Object {{ }}";
+						}
+					},
+				}
+			};
+			using (var b = CreateApkBuilder (Path.Combine ("temp", TestName))) {
+				Assert.IsTrue (b.Build (proj), "first build should have succeeded.");
+				className = "fOO";
+				proj.Touch ("Foo.cs");
+				Assert.IsTrue (b.Build (proj), "second build should have succeeded.");
```

Comment:
This should fail.. so why doers't it? We just changed the classname in the `Foo.cs` file. So it should still be generated in a `Foo.cs` not ` f00.cs`. I would expect a Java error in this second build.

Reply:
never mind. I get it now. The java will still generate a `f00.java` because of the class name, but the cs file remains the same.